File: src/lib/db/mongodb.ts
import { MongoClient, Db, Collection } from 'mongodb'

const MONGO_URL = process.env.MONGO_URL || 'mongodb://localhost:27017'
const DB_NAME = process.env.DB_NAME || 'wetnose_db'

let client: MongoClient | null = null
let db: Db | null = null

export async function connectToMongo(): Promise<Db> {
  if (db) return db
  
  try {
    client = new MongoClient(MONGO_URL)
    await client.connect()
    db = client.db(DB_NAME)
    console.log(`Connected to MongoDB: ${DB_NAME}`)
    return db
  } catch (error) {
    console.error('MongoDB connection error:', error)
    throw error
  }
}

export async function getDatabase(): Promise<Db> {
  if (!db) {
    await connectToMongo()
  }
  return db!
}

export async function getCollection<T>(name: string): Promise<Collection<T>> {
  const database = await getDatabase()
  return database.collection<T>(name)
}

export async function closeMongo(): Promise<void> {
  if (client) {
    await client.close()
    client = null
    db = null
  }
}

// Collection names - keep consistent with existing MongoDB collections
export const COLLECTIONS = {
  USERS: 'users',
  REFRESH_TOKENS: 'refresh_tokens',
  SERVICES: 'services',
  TEAM_MEMBERS: 'team_members',
  BLOG_POSTS: 'blog_posts',
  BLOG_CATEGORIES: 'blog_categories',
  GALLERY_ALBUMS: 'gallery_albums',
  GALLERY_ITEMS: 'gallery_items',
  TESTIMONIALS: 'testimonials',
  SETTINGS: 'settings',
  AUDIT_LOGS: 'audit_logs',
  APPOINTMENTS: 'appointments',
  CLINIC_RHYTHM: 'clinic_rhythm',
  POOL_QUESTIONS: 'pool_questions',
  POOL_ALARMS: 'pool_alarms',
  SLIDES: 'slides',
  SLIDER_SETTINGS: 'slider_settings',
  VERSIONS: 'versions',
  AI_GENERATIONS: 'ai_generations',
  CONTACT_MESSAGES: 'contact_messages',
  ROLE_PERMISSIONS: 'role_permissions'
}

File: backend/db/mongodb.py
from motor.motor_asyncio import AsyncIOMotorClient, AsyncIOMotorDatabase
from config import settings
import logging

logger = logging.getLogger(__name__)

class MongoDB:
    client: AsyncIOMotorClient = None
    db: AsyncIOMotorDatabase = None

mongodb = MongoDB()

async def connect_to_mongo():
    """Connect to MongoDB"""
    logger.info("Connecting to MongoDB...")
    mongodb.client = AsyncIOMotorClient(settings.MONGO_URL)
    mongodb.db = mongodb.client[settings.DB_NAME]
    logger.info(f"Connected to MongoDB database: {settings.DB_NAME}")

async def close_mongo_connection():
    """Close MongoDB connection"""
    logger.info("Closing MongoDB connection...")
    if mongodb.client:
        mongodb.client.close()
    logger.info("MongoDB connection closed")

def get_database() -> AsyncIOMotorDatabase:
    """Get database instance"""
    return mongodb.db
